CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs

```diff
@@ -115,7 +115,7 @@ public static class UploadToQnAMaker
                 Endpoint = $"https://{GetAppSetting("QnAServiceName")}.cognitiveservices.azure.com"
             };
 
-            string kbId = await GetKbID(log);
+            string kbId = await GetKbID(qnaClient, context, log);
             var updateKB = InitUpdateKB();
             var indexDocuments = new List<IndexDocument>();
             foreach (var msg in qnaQueueMessage.Values)
```

Comment:
You also need to filter out invalid file names

Reply:
added check for file extension and file size in upload-to-qna function